apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -69,7 +76,11 @@ export abstract class BaseConfiglessRushAction extends CommandLineAction {
       }
     }
 
-    if (!Utilities.shouldRestrictConsoleOutput()) {
+    if (!this._suppressStartupBanner && !Utilities.shouldRestrictConsoleOutput()) {
+      const banner: string | undefined = this._parser.startupBanner;
```

Comment:
Seems like it would be cleaner to move the banner generation code into the base action and only generate it if we're going to print it.

Reply:
Not a bad idea, I'll look at that.

Reply:
Updated.